src/test/tests/graphics/scenery/tests/unit/CurveGeometryTests.kt

```diff
@@ -13,58 +13,58 @@ import kotlin.test.assertNotNull
-        val z2 = Random.randomFromRange(-10f, 20f)
-        val x3 = Random.randomFromRange(20f, 30f)
-        val y3 = Random.randomFromRange(20f, 30f)
-        val z3 = Random.randomFromRange(20f, 30f)
-        val x4 = Random.randomFromRange(30f, 100f)
-        val y4 = Random.randomFromRange(30f, 100f)
-        val z4 = Random.randomFromRange(30f, 100f)
+        val point1 = Random.randomVectorFromRange(3, -30f, -10f)
+        val point2 = Random.randomVectorFromRange(3, -9f, 20f)
+        val point3 = Random.randomVectorFromRange(3, 21f, 30f)
+        val point4 = Random.randomVectorFromRange(3, 31f, 100f)
 
-        val controlPoints = arrayListOf(GLVector(x1, y1, z1), GLVector(x2, y2, z2), GLVector(x3, y3, z3),
-                GLVector(x4, y4, z4))
+        val controlPoints = arrayListOf(point1, point2, point3, point4)
 
         val curve = CatmullRomSpline(controlPoints)
 
         val geometry = CurveGeometry(curve)
         val frenetFrames = geometry.computeFrenetFrames(geometry.getCurve())
 
+        /*
+        For this baseShape function the number of points may differ each time
+        the baseShape function is invoked. However, the algorithm for calculating
+        the triangles only works if the number of points of the baseShape is constant
+        over the curve: the function should throw an error.
+         */
         fun triangleFalse(): ArrayList<GLVector> {
-            val i = Random.randomFromRange(9.5f, 10.5f)
+            val i = Random.randomFromRange(0.99f, 1.1f)
             val list = ArrayList<GLVector>()
             list.add(GLVector(0.3f, 0.3f, 0f))
             list.add(GLVector(0.3f, -0.3f, 0f))
             list.add(GLVector(-0.3f, -0.3f, 0f))
-            return if(i >= 10 ) {
+            return if(i >= 1 ) {
                 list
             } else {
                 list.add(GLVector(0f, 0f, 0f))
                 list
             }
         }
 
-
         assertEquals(curve.catMullRomChain(), geometry.getCurve())
         assertNotNull(frenetFrames.forEach { it.normal })
         assertNotNull(frenetFrames.forEach{ it.bitangent })
-        assertEquals(frenetFrames.filter { it.bitangent?.length2()!! < 1.1f && it.bitangent?.length2()!! > 0.9f },
+        assertEquals(frenetFrames.filter { it.bitangent?.length2()!! < 1.0001f && it.bitangent?.length2()!! > 0.99999f },
                 frenetFrames)
-        assertEquals(frenetFrames.filter { it.normal?.length2()!! < 1.1f && it.normal?.length2()!! > 0.9f },
+        assertEquals(frenetFrames.filter { it.normal?.length2()!! < 1.0001f && it.normal?.length2()!! > 0.99999f },
                 frenetFrames)
-        assertEquals(frenetFrames.filter { it.tangent.length2() < 1.1f && it.tangent.length2() > 0.9f },
+        assertEquals(frenetFrames.filter { it.tangent.length2() < 1.0001f && it.tangent.length2() > 0.99999f },
                 frenetFrames)
         assertFails {  geometry.drawSpline { triangleFalse() } }
```

Comment:
This test actually tests two things: Correct creation of Frenet frames, and failure of creation in case the number of vertices is not constant. Could you please split that into two tests that reflect those facts in their name? Thanks!

Reply:
Sure!

Reply:
Oh, do you mean two different classes or two different functions?